src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClient.cs

```diff
@@ -0,0 +1,359 @@
+            var downloads = ProcessRequest<DownloadsResponse>(downloadRequest);
+
+            foreach (var download in downloads.Downloads)
+            {
+
+                // If totalsize == 0 the torrent is a magnet downloading metadata
+                if (download.Size == null || download.Size == 0) continue;
+
+                var item = new DownloadClientItem();
+                item.DownloadId = InfoHash.FromHex(download.Infohash).ToHex();
+                //item.Category = Settings.TvCategory;
+                item.Title = download.Name;
+
+                item.DownloadClientInfo = DownloadClientItemClientInfo.FromDownloadClient(this);
+
+                // some concurrency could make this faster.
+                var filesRequest = getRequestBuilder("downloads/" + download.Infohash + "/files");
+                var files = ProcessRequest<GetFilesResponse>(filesRequest).Files;
+
+                item.OutputPath = new OsPath(download.Destination);
+                if (files.Count == 1)
+                {
+                    item.OutputPath += files.First().Name;
+                } else
+                {
+                    item.OutputPath += item.Title;
+                }
+
+                item.TotalSize = (long)download.Size;
+                item.RemainingSize = (long)(download.Size * (1 - download.Progress)); // TODO: i expect progress to be between 0 and 1
+                item.SeedRatio = download.Ratio;
+
+                if (download.Eta.HasValue)
+                {
+                    item.RemainingTime = TimeSpan.FromSeconds(download.Eta.Value);
+                }
+
+                //TODO: the item's message should not be equal to Error
+                item.Message = download.Error;
+
+                // tribler always saves files unencrypted to disk.
+                item.IsEncrypted = false;
+
+                // state handling
+
+                // TODO: impossible states?
+                //Failed = 4,
+                //Warning = 5
+
+
+                //Queued = 0,
+                //Completed = 3,
+                //Downloading = 2,
+
+                // Paused is guesstimated
+                //Paused = 1,
+
+
+                switch (download.Status)
+                {
+                    case DownloadStatus.DLSTATUS_HASHCHECKING:
+                    case DownloadStatus.DLSTATUS_WAITING4HASHCHECK:
+                    case DownloadStatus.DLSTATUS_DOWNLOADING:
+                        item.Status = DownloadItemStatus.Downloading;
+                        break;
+                    case DownloadStatus.DLSTATUS_METADATA:
+                    case DownloadStatus.DLSTATUS_ALLOCATING_DISKSPACE:
+                        item.Status = DownloadItemStatus.Queued;
+                        break;
+                    case DownloadStatus.DLSTATUS_SEEDING:
+                    case DownloadStatus.DLSTATUS_STOPPED:
+                        item.Status = DownloadItemStatus.Completed;
+                        break;
```

Comment:
Wouldn't hurt to add a default case and log out missing status handler as we do with qBit